### PR TITLE
[feature] NPC interaction dialog box

### LINKED ISSUE #129: Create NPC Interaction and Dialog Box
- Player cannot move (change input system)

### PR DESCRIPTION
**Pull request by uciii**
### Main changes
Create NPC interaction + slide up dialog box

### Tasks List
- [close #129]

### Attachments
https://user-images.githubusercontent.com/71260420/151897033-79d338fc-2f2c-4c87-8f74-4948590f8866.mov

### CODE REVIEW

**Review by RaiRiz — APPROVED** [review]
GG! However, I was thinking what if the animation of the appearance of the dialog box just appears (not slide up) so that it adds to the no-context impression and makes gaming experience more seamless (as in player does not need to wait for an animation to be finished first to quickly gain information)

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    [Tooltip("True if NPC is interacted")]
+    private bool _isInteracted = false;
+
+    public static event Action<bool> NpcInteractionEvent;
+    #endregion
+
+    public override void OnInteraction()
+    {
+        _isInteracted = !_isInteracted;
+        //Debug.Log("OnInteraction: " + _isInteracted);
```
Comment (VadimBhirawa) [inline Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:27]: If it works, no need to include comments.

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        if (_isInteracted)
+            NpcInteractionEvent?.Invoke(true);
+        else
+            NpcInteractionEvent?.Invoke(false);
+        //Debug.Log(
+        //    ("[INTERACTABLE] "+ this.name + (_isInteracted ? "on" : "off"))
```
Comment (VadimBhirawa) [inline Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:33]: This too

File: Assets/Astralization/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -105,4 +108,26 @@ private void MovePlayer()
...
+    }
+    #endregion
+
+    public void UpdatePlayerMovement(bool isInteractWithNpc)
+    {
+        if (isInteractWithNpc)
+            _playerInput.SwitchCurrentActionMap("Dialogue");
+        else
+            _playerInput.SwitchCurrentActionMap("Player");
+        //Debug.Log("isInteractWithNpc: " + isInteractWithNpc +
```
Comment (VadimBhirawa) [inline Assets/Astralization/Scripts/Player/PlayerMovement.cs:130]: And this comment too

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    #endregion
+
+    public void ShowDialogBox(bool isInteractWithNpc)
+    {
+        if (isInteractWithNpc)
+            _animator.SetBool("IsOpen", true);
+        else
+            _animator.SetBool("IsOpen", false);
+        //Debug.Log("isInteractWithNpc: " + isInteractWithNpc);
```
Comment (VadimBhirawa) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:45]: And this

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    [UnityTest]
+    public IEnumerator PlayerInteractableDetector_InteractNPC()
+    {
+        yield return new WaitWhile(() => sceneLoaded == false);
+        GameObject player = GameObject.Find("Party/Iris");
+        GameObject npc = GameObject.Find("NPC");
+        GameObject exclamationMark = npc.transform.Find("ExclamationMark").gameObject;
+        GameObject dialogue = GameObject.Find("Dialogue/Dialogue Box");
+
+        //Debug.Log(player + "||" + npc + "||" + exclamationMark + "||" + dialogue);
```
Comment (VadimBhirawa) [inline Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:28]: This too

**Review by VadimBhirawa — CHANGES_REQUESTED** [review]

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs (hunk 1 above)
Comment (uciii) [reply Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:27]: it's only used when someone wants to debug the program, so I think it's okay, wdyt?

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs (hunk 2 above)
Comment (uciii) [reply Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:33]: idem

File: Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs (hunk 5 above)
Comment (uciii) [reply Assets/Astralization/Tests/NpcTest/NpcInteractableTest.cs:28]: idem to all above

File: Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs (hunk 1 above)
Comment (VadimBhirawa) [reply Assets/Astralization/Scripts/Interactables/Controller/NpcController.cs:27]: alright seems good to me

**Review by VadimBhirawa — APPROVED** [review]
Looks good

**Pull request merged**